src/main/java/org/dbos/apiary/elasticsearch/ElasticsearchConnection.java

```diff
@@ -166,10 +168,14 @@ public boolean validate(Map<String, List<String>> writtenKeys, TransactionContex
         validationLock.unlock();
         if (valid) {
             for (String index : writtenKeys.keySet()) {
+                if (writtenKeys.get(index).size() > 10000) {
+                    continue; // Speed up bulk-loading in benchmarks.
+                }
                 for (String key : writtenKeys.get(index)) {
                     while (true) {
                         try {
-                            client.updateByQuery(ubq -> ubq
+                            long t0 = System.nanoTime();
+                            UpdateByQueryResponse r = client.updateByQuery(ubq -> ubq
                                     .index(index)
                                     .query(BoolQuery.of(bb -> bb
                                             .must(
```

Comment:
For the update filter here, could you simply only update old records with `endVersion == infinity` (add one more term query)? So it should only update the last version instead of updating every previous version. I think it can still guarantee correctness.

Reply:
At large scale that filter is more expensive than the additional updates